iati/core/rulesets.py

```diff
@@ -67,35 +63,38 @@ def __init__(self, ruleset_str):
 
         Raises:
             TypeError: When a ruleset_str is not a string.
-            ValueError: When ruleset_str does not validate against the ruleset schema.
-
-        Todo:
-            Raises a UnicodeDecodeError or json.JSONDecodeError if passed a dodgey bytearray in all Python versions except 3.6.
+            ValueError: When a ruleset_str does not validate against the ruleset schema or cannot be correctly decoded.
 
         """
-        self.ruleset = json.loads(ruleset_str, object_pairs_hook=iati.core.utilities.dict_raise_on_duplicates)
+        try:
+            self.ruleset = json.loads(ruleset_str, object_pairs_hook=iati.core.utilities.dict_raise_on_duplicates)
+        except TypeError:
+            raise ValueError
         self.validate_ruleset()
         self.rules = set()
-        self.set_rules()
+        self._set_rules()
 
     def validate_ruleset(self):
-        """Validate a Ruleset against the Ruleset Schema."""
+        """Validate a Ruleset against the Ruleset Schema.
+
+        Raises:
+            ValueError: When ruleset_str does not validate against the ruleset schema.
+
+        """
         try:
             jsonschema.validate(self.ruleset, iati.core.default.ruleset_schema())
         except jsonschema.ValidationError:
             raise ValueError
 
-    def set_rules(self):
+    def _set_rules(self):
         """Set the Rules of the Ruleset."""
```

Comment:
There are no args or return type. As such, it is somehat unclear from the docstring alone what this function is doing. Suggest an expansion of the docstring (while avoiding implementation details!)